birdseye/utils.py

```diff
@@ -32,7 +32,16 @@ def all_file_paths():
 
 
 def short_path(path):
-    return strip_required_prefix(path, os.path.commonprefix(all_file_paths())) or path_leaf(path)
+    """
+    Return a shortened version of the received path, by removing sections
+    of the path common to other filepaths in the db.
+    """
+    # We only use the directory portion when getting a common prefix, to avoid
+    # stripping from the path's filename.
+    all_file_dirs = [ntpath.dirname(f) for f in all_file_paths()]
```

Comment:
This is a good idea, but I don't think it solves the problem in general. You could also have two directories which share a common prefix in the name and this would cut the directory name in half.

Reply:
Yes, I should have picked that up. 

I've updated with 2 commits and tested with some directory names which are partially similar.